### PR TITLE
Add Pagination to claims and create Pagination component

### PR DESCRIPTION
**Pull request by CatalinVoineag**
## Context

Some index pages don't have paginations and there is some logic around them so I thought to create a component for it.

## Changes proposed in this pull request

Pagination Component
Claims Index pages

## Guidance to review

Pull this on local
Change pagination records per page in the pagy initializer
Check all index pages have pagination

## Screenshots

Screenshot from 2024-03-11 16-44-36

### CODE REVIEW

**Comment by CatalinVoineag** [comment]
[Add pagination to all list pages in Claims](https://trello.com/c/UlhjoDPz/205-add-pagination-to-all-list-pages-in-claims)

**Review by JamieCleare2525 — APPROVED** [review]

File: spec/components/pagination_component_spec.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+require "rails_helper"
+
+RSpec.describe PaginationComponent, type: :component do
+  it "renders the pagination content" do
+    pagy = Pagy.new(count: 100, page: 2)
+    render_inline described_class.new(pagy:)
+
+    within "govuk-pagination__list" do
```
Comment (ollietreend) [inline spec/components/pagination_component_spec.rb:8]: This feels like a silly question, but.. why does this work? 🤔

I'm assuming `govuk-pagination__list` is a class on a HTML element. So I'd have expected it to start with a `.` to signify that it's a CSS class selector?

```suggestion
    within ".govuk-pagination__list" do
```

File: spec/components/pagination_component_spec.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    end
+    expect(page).to have_content("Showing 26 to 50 of 100 results")
+  end
+
+  context "when we don't have more than 1 page of pagination" do
+    it "doesn't render the pagination content" do
+      pagy = Pagy.new(count: 25, page: 1)
+      render_inline described_class.new(pagy:)
+
+      within "govuk-pagination__list" do
```
Comment (ollietreend) [inline spec/components/pagination_component_spec.rb:21]: ..and here

**Review by ollietreend — COMMENTED** [review]
Sorry I didn't get this in before you merged it @CatalinVoineag – but this particular thing has confused me.. 🤔

File: spec/components/pagination_component_spec.rb (hunk 1 above)
Comment (CatalinVoineag) [reply spec/components/pagination_component_spec.rb:8]: I'll get this fixed in my next PR

**Pull request merged**